### PR TITLE
Create base UI page, add step and features files and initial work for BDD

### LINKED ISSUE #77: Create a .feature and the accompanying steps.py files
**As a** Developer
**I need** a .feature and the accompanying steps.py files
**So that** the RESTful APIs can be tested from the outside-in.

 ### Assumptions
- 'behave' is being used to execute Selenium
- Gherkin syntax is used
- There would be two steps.py files: web_steps.py and customer_steps.py
- customer_steps.py: For loading customer data
- web_steps.py: For the BDD tests.

### Acceptance Criteria
 ```gherkin
 Given BDD is being employed
 When the command behave is run
 Then BDD tests are executed
 ```

When being updated, the Steps.py files should be of the following syntax:
```
# steps.py
from behave import given, when, then

@given('some known state')
def step_impl(context):
  set_up(some, state)

@when('some action is taken')
def step_impl(context):
  perform(some, action)

@then('some outcome is observed')
def step_impl(context):
  assert(outcome, observed)
```

### LINKED ISSUE #83: Create a simple UI to administer the app
**As a** Site Manager
**I need** to be able to interact with a simple UI
**So that** I can manage the customers' data

 ### Assumptions
- We have the IP and Port details

### Acceptance Criteria
 ```gherkin
 Given the application is running
 When I navigate to the app's root IP and Port in a browser
 Then a simple UI should be presented to me
 ```

### PR DESCRIPTION
**Pull request by TestSubjector**
- Updated ```requirements.txt```, please rebuild your docker container
- Created the Basic UI page (closes #83 ), can be accessed on ```localhost``` using ```honcho start```
- Created the required features and steps.py files (closes #77 ), can be tested using ```behave```
- Initial work for BDD workflow 

Important notes:
For future work, after creating any UI functionality, you'll need to 
- Update rest_api.js (in ```service/static/js/rest_api.js```) 
- Got to customers.feature(```features/customers.feature```), and add write BDD for the implemented functionality
- To test the BDD, create a terminal and run ```honcho start```. In a separate terminal run ```behave```

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -65,37 +65,39 @@ def init_db():
...
-                },
-                "List all customers": {
-                    "method": "GET",
-                    "url": url_for("list_customers", _external=True)
-                }
-            }
-        ),
-        status.HTTP_200_OK,
-    )
+    # """Root URL response"""
```
Comment (mm12063) [inline service/routes.py]: Is it necessary for these to be commented out instead of removed entirely if we're not using it?

**Comment by mm12063** [comment]
Just FYI, going forward we need to make sure it's just one PR per Issue to keep things clean.

File: service/routes.py (hunk 1 above)
Comment (TestSubjector) [reply service/routes.py]: Removed the commented code

**Comment by TestSubjector** [comment]
> Just FYI, going forward we need to make sure it's just one PR per Issue to keep things clean.

Sure, will make sure of that. This was just a bit tricky case (making sure the ```static``` code was correct required the feature and step files)

**Pull request merged**